### PR TITLE
Make `next/prev` infer `--edit` from running on non-head

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/prev.rs [hunk 1]
```diff
@@ -44,6 +44,9 @@ use crate::ui::Ui;
 /// B    B
 /// |    |
 /// A    A
+///
+/// If your working-copy commit is already a non-head commit (i.e. if it has
+/// visible children), then `--edit` is implied.
```
Comment (PhilipMetzger) [inline cli/src/commands/prev.rs:49]: This is also bad copypasta, shouldn't it be parents here? 

I assume our users will stumble over this terminology. Can we just simplify it to "If you working-commit is already a commit with a child, then `--edit` is implied".

File: cli/src/commands/next.rs [hunk 2]
```diff
@@ -48,6 +48,9 @@ use crate::ui::Ui;
 /// B => @
 /// |    |
 /// @    A
+///
+/// If your working-copy commit is already a non-head commit (i.e. if it has
+/// visible children), then `--edit` is implied.
```
Comment (PhilipMetzger) [inline cli/src/commands/next.rs:53]: Ditto.

**Review by PhilipMetzger — APPROVED** [review]
LGTM, still pondering if we should make it configurable.

File: cli/src/commands/prev.rs (hunk 1 above)
Comment (martinvonz) [reply cli/src/commands/prev.rs]: > This is also bad copypasta, shouldn't it be parents here?

No, it's still the same condition for inferring that you want `--edit`.

> I assume our users will stumble over this terminology. Can we just simplify it to "If you working-commit is already a commit with a child, then `--edit` is implied".

True, it's better to use simpler words. Fixed.

Comment (PhilipMetzger) [reply cli/src/commands/prev.rs]: > > This is also bad copypasta, shouldn't it be parents here?
> 
> No, it's still the same condition for inferring that you want `--edit`.

I understand what you mean in terms of graph terminology. But again, someone who is not aware will expect it to be parents.

Comment (martinvonz) [reply cli/src/commands/prev.rs]: But it would be incorrect to say parents even if they would expect that :) What do you suggest we write instead?

Comment (PhilipMetzger) [reply cli/src/commands/prev.rs]: > What do you suggest we write instead?

Not sure. 🤷 I still there is a terminolog missmatch.

Comment (martinvonz) [reply cli/src/commands/prev.rs]: I'll merge this as is then. If it's confusing users, we will surely hear about it eventually and maybe one of those users have a suggestion.

**Comment by emesterhazy** [comment]
The problem I have with this is that if I `jj new` an interior commit and then `jj squash` it, the working copy will be changed to the commit that I squashed my changes into. As a result, `jj next` and `jj prev` don't work as I might hope.

But, you can always run `jj new` after so it's not a huge issue.

**Pull request merged**